linkschkr_test.go

```diff
@@ -0,0 +1,69 @@
+package linkschkr
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestCrawler(t *testing.T) {
+	t.Parallel()
+
+	nWorkers := 1000 // just for fun
+	work := make(chan *Work)
+	results := make(chan *Result)
+
+	for n := 0; n < nWorkers; n++ {
+		go Crawler(n, work)
+	}
+
+	body := `<html>
```

Comment:
It might make sense to put this HTML in a `testdata` file, and we could have a bunch of them, ideally taken from real websites, so that we can test whether we successfully parse all the links out of them.